packaging/osx/buildpackage.sh

```diff
@@ -12,12 +12,12 @@ fi
 LAUNCHER_TAG="osx-launcher-20180723"
 
 if [ $# -ne "2" ]; then
-	echo "Usage: `basename $0` tag outputdir"
+	echo "Usage: $(basename "$0") tag outputdir"
     exit 1
 fi
 
 # Set the working dir to the location of this script
-cd $(dirname $0)
+cd "$(dirname "$0")" || exit
```

Comment:
Wouldn't it be better to use `exit 1` here to indicate an error?

Reply:
Exiting with the correct error code will let the parent detect that this failed, and respond appropriately.

Reply:
Ah OK, thanks for telling me I'll implement this then.

Reply:
Should I implement this everywhere `|| exit` appears in these scripts?

Reply:
Looks like the preexisting cases already specified an error code, so yes please implement on all the new additions.

Reply:
I do believe in 7bee2740b I have implemented your wishes.

Reply:
Good start, but it should be done for `pushd` and `popd` as well (basically everywhere the `|| exit` construct is used).

Reply:
So that includes where `> /dev/null` occurs after the pushd and before the `|| exit`? Those lines were there before this PR.

Reply:
As I'm about to go to sleep and I'm pretty sure you mean those lines too I'm going to push that change.